public/app/features/explore/TraceView/TraceView.tsx

```diff
@@ -50,7 +56,16 @@ export function TraceView(props: Props) {
     detailTagsToggle,
     detailWarningsToggle,
     detailStackTracesToggle,
+    clearDetailStates,
   } = useDetailState();
+
+  // Clear detail state when new trace arrives
+  useEffect(() => {
+    if (frame && prevFrame && !isEqual(prevFrame, frame)) {
+      clearDetailStates();
+    }
+  }, [frame, prevFrame, clearDetailStates]);
```

Comment:
I think we could just have `frame` as dependency can't we? I mean if we don't need deep equal here, then the dependency array is checked whether it changes and basically giving you the `frame != prevFrame` check out of the box.

Reply:
Also maybe it could make sense to put this effect inside `useDetailState`